rocks/RocksDatabase.java

```diff
@@ -302,10 +337,21 @@ void remove(RocksSession session) {
 
     protected void close() {
         if (isOpen.compareAndSet(true, false)) {
+            if (rocksPropertiesLogger != null) shutdownRocksPropertiesLogger();
             closeResources();
         }
     }
 
+    private void shutdownRocksPropertiesLogger() {
+        try {
+            rocksPropertiesLogger.shutdown();
+            boolean terminated = rocksPropertiesLogger.awaitTermination(5, SECONDS);
+            if (!terminated) throw TypeDBException.of(ROCKS_LOGGER_SHUTDOWN_FAILED);
+        } catch (InterruptedException e) {
+            throw TypeDBException.of(e);
+        }
+    }
+
```

Comment:
if the user tells the `rocks` package to `debug`, then we periodically log properties we read from RocksDB. Really useful for debugging open snapshots, and getting estimated data size/number of keys, etc.